cristin-organization/src/main/java/no/unit/nva/cristin/organization/QueryCristinOrganizationHandler.java

```diff
@@ -0,0 +1,142 @@
+package no.unit.nva.cristin.organization;
+
+
+import com.amazonaws.services.lambda.runtime.Context;
+import no.unit.nva.cristin.common.Utils;
+import no.unit.nva.cristin.common.model.SearchResponse;
+import no.unit.nva.cristin.common.util.UriUtils;
+import no.unit.nva.cristin.model.Organization;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.BadRequestException;
+import nva.commons.core.JacocoGenerated;
+
+import java.net.HttpURLConnection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static no.unit.nva.cristin.projects.Constants.DEFAULT_NUMBER_OF_RESULTS;
+import static no.unit.nva.cristin.projects.Constants.FIRST_PAGE;
+import static no.unit.nva.cristin.projects.Constants.LANGUAGE;
+import static no.unit.nva.cristin.projects.Constants.NUMBER_OF_RESULTS;
+import static no.unit.nva.cristin.projects.Constants.PAGE;
+import static no.unit.nva.cristin.projects.Constants.QUERY;
+import static no.unit.nva.cristin.projects.ErrorMessages.ERROR_MESSAGE_INVALID_QUERY_PARAMS_ON_LOOKUP;
+import static no.unit.nva.cristin.projects.ErrorMessages.ERROR_MESSAGE_LANGUAGE_INVALID;
+import static no.unit.nva.cristin.projects.ErrorMessages.ERROR_MESSAGE_NUMBER_OF_RESULTS_VALUE_INVALID;
+import static no.unit.nva.cristin.projects.ErrorMessages.ERROR_MESSAGE_PAGE_VALUE_INVALID;
+import static no.unit.nva.cristin.projects.ErrorMessages.ERROR_MESSAGE_QUERY_MISSING_OR_HAS_ILLEGAL_CHARACTERS;
+import static nva.commons.core.attempt.Try.attempt;
+
+@SuppressWarnings("unused")
+public class QueryCristinOrganizationHandler extends ApiGatewayHandler<Void, SearchResponse<Organization>> {
+
+    protected static final String DEFAULT_LANGUAGE_CODE = "nb";
+    private static final char CHARACTER_DASH = '-';
+    private static final char CHARACTER_COMMA = ',';
+    private static final char CHARACTER_PERIOD = '.';
+    private static final Set<String> VALID_QUERY_PARAMS = Set.of(QUERY, LANGUAGE, PAGE, NUMBER_OF_RESULTS);
+    private static final Set<String> VALID_LANGUAGE_CODES = Set.of("en", "nb", "nn");
+    private final transient CristinApiClient cristinApiClient;
+
+    @JacocoGenerated
+    public QueryCristinOrganizationHandler() {
+        this(new CristinApiClient());
+    }
+
+    public QueryCristinOrganizationHandler(CristinApiClient cristinApiClient) {
+        super(Void.class);
```

Comment:
You can use an Environement object in constructor, but its not needed at the moment it seems